pkg/controller/federation/clusterdeployment_federation_controller.go

```diff
@@ -172,14 +185,15 @@ func (r *ReconcileClusterDeploymentFederation) Reconcile(request reconcile.Reque
 
 	err = kubefed2.JoinCluster(hostConfig,
 		targetConfig,
-		federationutil.DefaultFederationSystemNamespace,
-		federationutil.MulticlusterPublicNamespace,
+		cd.Namespace,
+		cd.Namespace,
```

Comment:
Slightly confused, it sounded from your comments in main PR like we had to start federating them into the central namespaces and couldn't use the cd.Namespace itself?